cpp/perspective/src/cpp/vocab.cpp

```diff
@@ -73,9 +73,9 @@ t_vocab::get_interned(const char* s) {
     t_sidxmap::iterator iter = m_map.find(s);
 
     t_uindex idx, bidx, eidx;
-    t_uindex len = strlen(s) + 1;
 
     if (iter == m_map.end()) {
+        t_uindex len = strlen(s) + 1;
```

Comment:
could save this strlen() for callers which already know the string's length (e.g. anything with a `std::string`) by providing an overload which takes a `size_t s_len`

Reply:
Good point - but for the records, in this instance `t_vocab` is only used for values that can be `t_scalar`, which does not include `std::string`.